Adds access control lists which differentiate between
LAN and non-LAN IP-adresses as well as between the
to-localhost and from-localhost packet flow direction.

Index: squid/squid.conf
===================================================================
--- squid.orig/squid.conf
+++ squid/squid.conf
@@ -4873,11 +4873,16 @@
 # list processing. Be warned that this can be quite some lines per request.
 
 acl manager proto cache_object
-acl localhost src 127.0.0.1/32
-acl to_localhost dst 127.0.0.0/8 0.0.0.0/32
-acl localnet src 10.0.0.0/8	# RFC1918 possible internal network
-acl localnet src 172.16.0.0/12	# RFC1918 possible internal network
-acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
+acl from_localhost src 127.0.0.1/32
+acl to_localhost dst 127.0.0.0/8
+acl to_lan dst 192.168.0.0/16
+acl to_lan dst 10.0.0.0/8
+acl to_lan dst 172.16.0.0/12
+acl to_lan dst 169.254.0.0/16
+acl from_lan src 192.168.0.0/16
+acl from_lan src 10.0.0.0/8
+acl from_lan src 172.16.0.0/12
+acl from_lan src 169.254.0.0/16
 
 acl SSL_ports port 443
 acl Safe_ports port 80		# http
@@ -4894,17 +4899,19 @@ acl Safe_ports port 901		# SWAT
 acl purge method PURGE
 acl CONNECT method CONNECT
 
-http_access allow manager localhost
+http_access allow manager from_localhost
 http_access deny manager
+http_access allow ssh_ports
+http_access deny to_localhost
 http_access allow purge localhost
 http_access deny purge
-http_access deny !Safe_ports
+http_access deny !Safe_ports !SSL_ports
 http_access deny CONNECT !SSL_ports
-http_access allow localnet
-http_access allow localhost
 http_access deny all
 icp_access deny all
 htcp_access deny all
+  
+always_direct allow from_localhost to_localhost
 
 http_port 3128
 
